kura/distrib/src/main/resources/common/customize_iptables.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2024 Eurotech and/or its affiliates and others
+#
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#  Eurotech
+#
+import sys
+import logging
+import os
+from network_tools import get_eth_wlan_interfaces_names
+
+def main():
+    logging.basicConfig(
+        format='[customize_iptables.py] %(asctime)s %(levelname)s %(message)s',
+        level=logging.INFO,
+        datefmt='%Y-%m-%d %H:%M:%S',
+        handlers=[
+            logging.StreamHandler()
+        ]
+    )
+    
+    iptables_filename = "/opt/eclipse/kura/.data/iptables"
+    
+    (eth_names, wlan_names) = get_eth_wlan_interfaces_names()
+    
+    eth_number = len(eth_names)
+    wlan_number = len(wlan_names)
+    
+    if eth_number == 0:
+        logging.info("ERROR: no ethernet interface found")
+        exit(1)
+    
+    logging.info("%s : starting editing", iptables_filename)
+    iptables = open(iptables_filename, 'r+', encoding='utf-8')
+    iptables_content = ""
+    if wlan_number == 0:
+        for line in iptables:
+            if 'WIFI_INTERFACE_0' not in line:
+                iptables_content += line.replace('ETH_INTERFACE_0', eth_names[0])
+    else:
+        iptables_content = iptables.read()
+        iptables_content = iptables_content.replace('ETH_INTERFACE_0', eth_names[0])
+        iptables_content = iptables_content.replace('WIFI_INTERFACE_0', wlan_names[0])
+    
+    iptables.seek(0)
+    iptables.truncate()
+    iptables.write(iptables_content)
+    iptables.close()
```

Comment:
Prefer the use of the `with` statement when working with files. See: https://www.freecodecamp.org/news/with-open-in-python-with-statement-syntax-example/. In case of exceptions being thrown the file gets closed for you automatically.

Also you can simplify the code a little to improve readability

```suggestion
    with open(iptables_filename, 'r', encoding='utf-8') as iptables:
        iptables_content = iptables.read()
    
    if wlan_number == 0:
        for line in iptables_content.split("\n"):
            if 'WIFI_INTERFACE_0' not in line:
                iptables_content += line.replace('ETH_INTERFACE_0', eth_names[0])
    else:
        iptables_content = iptables_content.replace('ETH_INTERFACE_0', eth_names[0])
        iptables_content = iptables_content.replace('WIFI_INTERFACE_0', wlan_names[0])
    
    with open(iptables_filename, 'w', encoding='utf-8') as iptables:
        iptables.write(iptables_content)
```

Reply:
In case you don't want the to `split` the `iptables_content` you can read the file and put its content in an array of strings like so:

```python
with open(iptables_filename, 'r', encoding='utf-8') as iptables:
   iptables_content = [line for line in iptables]
```